src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -358,16 +388,42 @@ def process_messages(
 
     org_strings = defaultdict(set)
     strings = set()
+    skipped_offsets = set()
     with metrics.timer("process_messages.parse_outer_message"):
         parsed_payloads_by_offset = {
             msg.offset: json.loads(msg.payload.value.decode("utf-8"), use_rapid_json=True)
             for msg in outer_message.payload
         }
-        for message in parsed_payloads_by_offset.values():
+        for offset, message in parsed_payloads_by_offset.items():
             metric_name = message["name"]
             org_id = message["org_id"]
             tags = message.get("tags", {})
 
+            if not valid_metric_name(metric_name):
+                logger.warning(
+                    "process_messages.invalid_metric_name",
+                    extra={"org_id": org_id, "metric_name": metric_name, "offset": offset},
+                )
+                skipped_offsets.add(offset)
+                continue
+
+            is_valid, invalid_strs = valid_metric_tags(tags)
+
+            if not is_valid:
+                # sentry doesn't seem to actually capture nested logger.error extra args
```

Comment:
I think logger args in sentry in extra go into structured logs in GCP where they can be searched by attribute. @volokluev knows more about this. Unfortunately we don't have this configured for snuba